### PR TITLE
Fix relative URLs having a null argument after resolution to `LinkDetails`

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

**Comment by smoogipoo** [comment]
Should the one on line 173 also be updated? Seems like it wouldn't handle `https://google.com` if I'm reading the surrounding code correctly.

**Comment by peppy** [comment]
Actually I'm not sure any more... Maybe we should be using the `Url` not the `Argument` in the first place? Will check.

**Comment by peppy** [comment]
As per discord conversation, it's not worth looking into this too deeply. I've just made all `External` types now get the argument.

**Review by smoogipoo — DISMISSED** [review]

File: osu.Game/Graphics/Containers/LinkFlowContainer.cs [hunk 1]
```diff
@@ -70,7 +70,7 @@ public void AddLink(LocalisableString text, LinkAction action, string argument,
             createLink(spriteText.Yield(), new LinkDetails(action, argument), tooltipText);
         }
 
-        public void AddLink(IEnumerable<SpriteText> text, LinkAction action = LinkAction.External, string linkArgument = null, string tooltipText = null)
+        public void AddLink(IEnumerable<SpriteText> text, LinkAction action, string linkArgument, string tooltipText = null)
```
Comment (bdach) [inline osu.Game/Graphics/Containers/LinkFlowContainer.cs:73]: Not 100% sure what the removal of the default on `LinkAction` is aiming to achieve, but it does look to be more consistent with the other methods signature-wise.

File: osu.Game/Graphics/Containers/LinkFlowContainer.cs [hunk 2]
```diff
@@ -57,7 +57,7 @@ public void AddLinks(string text, List<Link> links)
             createLink(AddText(text, creationParameters), new LinkDetails(LinkAction.External, url), url);
 
         public void AddLink(string text, Action action, string tooltipText = null, Action<SpriteText> creationParameters = null)
-            => createLink(AddText(text, creationParameters), new LinkDetails(LinkAction.Custom, null), tooltipText, action);
+            => createLink(AddText(text, creationParameters), new LinkDetails(LinkAction.Custom, @"action"), tooltipText, action);
```
Comment (bdach) [inline osu.Game/Graphics/Containers/LinkFlowContainer.cs:60]: Could alternatively use `string.Empty` here, maybe, for a slightly less magic value?

**Review by bdach — COMMENTED** [review]
I could be fine with this as is, just gonna throw in a few loose comments.

File: osu.Game/Graphics/Containers/LinkFlowContainer.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Graphics/Containers/LinkFlowContainer.cs:73]: The main change here was removing the `null` portion of `linkArgument`. There's only one usage of this function though, and it felt like explicitly requiring the action type is better.

File: osu.Game/Graphics/Containers/LinkFlowContainer.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Graphics/Containers/LinkFlowContainer.cs:60]: Yeah, almost changed this myself after noticing how other use cases use the value. Will update.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**